lib/ansible/modules/cloud/amazon/ec2_group.py

```diff
@@ -284,6 +295,31 @@ def get_target_from_rule(module, ec2, rule, name, group, groups, vpc_id):
     return group_id, ip, target_group_created
 
 
+def ensure_tags(ec2, group, tags, changed, check_mode):
+    try:
+        # Name tag is empty per default. To avoid unnecessary updates
+        if tags is None:
+            tags = {"Name": ""}
+        resource_id = group.id
+        cur_tags = group.tags
+        if cur_tags == tags:
+            return changed
+
+        to_delete = dict((k, cur_tags[k]) for k in cur_tags if k not in tags)
+        if to_delete:
+            ec2.delete_tags(resource_id, to_delete, dry_run=check_mode)
+
+        to_add = dict((k, tags[k]) for k in tags if k not in cur_tags or cur_tags[k] != tags[k])
+        if to_add:
+            ec2.create_tags(resource_id, to_add, dry_run=check_mode)
+
+        # latest_tags = get_resource_tags(vpc_conn, resource_id)
+        return True
+    except EC2ResponseError as e:
+        raise AnsibleTagCreationException(
```

Comment:
I think you need to define `AnsibleTagCreationException` somewhere for this to work.